iiod/iiod.c

```diff
@@ -459,40 +469,77 @@ int main(int argc, char **argv)
 		}
 	}
 
-	ctx = iio_create_context_from_uri(arg);
-
-	if (!ctx) {
+	main_thread_pool = thread_pool_new();
+	if (!main_thread_pool) {
 		iio_strerror(errno, err_str, sizeof(err_str));
-		IIO_ERROR("Unable to create local context: %s\n", err_str);
+		IIO_ERROR("Unable to create thread pool: %s\n", err_str);
 		return EXIT_FAILURE;
 	}
 
-	xml_zstd = get_xml_zstd_data(ctx, &xml_zstd_len);
+	if (WITH_IIOD_USBD && ffs_mountpoint) {
+		ret = init_usb_daemon(ffs_mountpoint, nb_pipes);
+		if (ret < 0) {
+			iio_strerror(errno, err_str, sizeof(err_str));
+			IIO_ERROR("Unable to init USB: %s\n", err_str);
 
-	main_thread_pool = thread_pool_new();
-	if (!main_thread_pool) {
-		iio_strerror(errno, err_str, sizeof(err_str));
-		IIO_ERROR("Unable to create thread pool: %s\n", err_str);
-		ret = EXIT_FAILURE;
-		goto out_destroy_context;
+			thread_pool_destroy(main_thread_pool);
+			return EXIT_FAILURE;
+		}
+
+		ep0_fd = ret;
 	}
 
 	set_handler(SIGHUP, sig_handler);
 	set_handler(SIGPIPE, sig_handler);
 	set_handler(SIGINT, sig_handler);
 	set_handler(SIGTERM, sig_handler);
+	set_handler(SIGUSR1, sig_handler_usr1);
+
+	do {
+		thread_pool_restart(main_thread_pool);
+		restart_usr1 = false;
+
+		ret = start_iiod(arg, ffs_mountpoint, uart_params, debug,
+				 interactive, use_aio, port, nb_pipes, ep0_fd);
+	} while (!ret && restart_usr1);
+
+	thread_pool_destroy(main_thread_pool);
+	close(ep0_fd);
```

Comment:
is this on purpose? In case we don't get into `if (WITH_IIOD_USBD && ffs_mountpoint) `, won't we close stdin in here (being this a daemon it's likely already closed anyways)?

I'm fairly sure it's not really a problem but for correctness shouldn't we do something like:

```
if (WITH_IIOD_USBD && ffs_mountpoint)
    close(ep0_fd);
```

Reply:
It's closed just before the main() exits, so it shouldn't really have any side effects. But yes, you are correct, that's an oversight.